File: swift/Tests/EnsoulTests/ConfigTests.swift
/// Tests for ClientConfig — defaults, custom values, and derived apiURL.
import XCTest
@testable import Ensoul

final class ConfigTests: XCTestCase {

    // MARK: - Default values

    func test_config_defaultBaseURL() {
        let config = ClientConfig()
        XCTAssertEqual(config.baseURL, "https://api.ensoul-ai.com")
    }

    func test_config_defaultTimeout() {
        let config = ClientConfig()
        XCTAssertEqual(config.timeout, 30)
    }

    func test_config_defaultMaxRetries() {
        let config = ClientConfig()
        XCTAssertEqual(config.maxRetries, 2)
    }

    func test_config_defaultAPIKeyIsNil() {
        let config = ClientConfig()
        XCTAssertNil(config.apiKey)
    }

    func test_config_defaultBearerTokenIsNil() {
        let config = ClientConfig()
        XCTAssertNil(config.bearerToken)
    }

    func test_config_defaultCustomHeadersIsEmpty() {
        let config = ClientConfig()
        XCTAssertTrue(config.customHeaders.isEmpty)
    }

    // MARK: - Custom values

    func test_config_customAPIKey() {
        let config = ClientConfig(apiKey: "ens_test_key")
        XCTAssertEqual(config.apiKey, "ens_test_key")
    }

    func test_config_customBearerToken() {
        let config = ClientConfig(bearerToken: "eyJhbGci.payload.sig")
        XCTAssertEqual(config.bearerToken, "eyJhbGci.payload.sig")
    }

    func test_config_customBaseURL() {
        let config = ClientConfig(baseURL: "https://staging.ensoul.ai")
        XCTAssertEqual(config.baseURL, "https://staging.ensoul.ai")
    }

    func test_config_customTimeout() {
        let config = ClientConfig(timeout: 60)
        XCTAssertEqual(config.timeout, 60)
    }

    func test_config_customMaxRetries() {
        let config = ClientConfig(maxRetries: 5)
        XCTAssertEqual(config.maxRetries, 5)
    }

    func test_config_customHeaders() {
        let config = ClientConfig(customHeaders: ["X-Tenant": "acme"])
        XCTAssertEqual(config.customHeaders["X-Tenant"], "acme")
    }

    // MARK: - apiURL derived property

    func test_config_apiURL_appendsVersionToBaseURL() {
        let config = ClientConfig(baseURL: "https://api.ensoul-ai.com")
        XCTAssertEqual(config.apiURL, "https://api.ensoul-ai.com/v1")
    }

    func test_config_apiURL_stripsTrailingSlash() {
        let config = ClientConfig(baseURL: "https://api.ensoul-ai.com/")
        XCTAssertEqual(config.apiURL, "https://api.ensoul-ai.com/v1")
    }

    func test_config_apiURL_stripsMultipleTrailingSlashes() {
        let config = ClientConfig(baseURL: "https://api.ensoul-ai.com///")
        XCTAssertEqual(config.apiURL, "https://api.ensoul-ai.com/v1")
    }

    func test_config_apiURL_customBaseURL() {
        let config = ClientConfig(baseURL: "https://staging.ensoul.ai")
        XCTAssertEqual(config.apiURL, "https://staging.ensoul.ai/v1")
    }

    func test_config_apiURL_localhostBaseURL() {
        let config = ClientConfig(baseURL: "http://localhost:8000")
        XCTAssertEqual(config.apiURL, "http://localhost:8000/v1")
    }
}

File: unity/Tests/ConfigTests.cs
using System;
using System.Collections.Generic;
using NUnit.Framework;

namespace Ensoul.Tests
{
    [TestFixture]
    public class ConfigTests
    {
        [Test]
        public void DefaultConfig_HasCorrectDefaults()
        {
            var config = new EnsoulConfig();
            Assert.That(config.BaseUrl, Is.EqualTo("https://api.ensoul-ai.com"));
            Assert.That(config.Timeout, Is.EqualTo(TimeSpan.FromSeconds(30)));
            Assert.That(config.MaxRetries, Is.EqualTo(2));
            Assert.That(config.ApiKey, Is.Null);
            Assert.That(config.BearerToken, Is.Null);
        }

        [Test]
        public void CustomConfig_StoresAllValues()
        {
            var config = new EnsoulConfig(
                baseUrl: "https://staging.ensoul.ai",
                apiKey: "key-123",
                bearerToken: "token-abc",
                timeout: TimeSpan.FromSeconds(60),
                maxRetries: 5,
                customHeaders: new Dictionary<string, string> { ["X-Custom"] = "value" });

            Assert.That(config.BaseUrl, Is.EqualTo("https://staging.ensoul.ai"));
            Assert.That(config.ApiKey, Is.EqualTo("key-123"));
            Assert.That(config.BearerToken, Is.EqualTo("token-abc"));
            Assert.That(config.Timeout, Is.EqualTo(TimeSpan.FromSeconds(60)));
            Assert.That(config.MaxRetries, Is.EqualTo(5));
            Assert.That(config.CustomHeaders["X-Custom"], Is.EqualTo("value"));
        }

        [Test]
        public void ApiUrl_AppendsVersion()
        {
            var config = new EnsoulConfig(baseUrl: "https://api.ensoul-ai.com");
            Assert.That(config.ApiUrl, Is.EqualTo("https://api.ensoul-ai.com/v1"));
        }

        [Test]
        public void ApiUrl_StripsTrailingSlash()
        {
            var config = new EnsoulConfig(baseUrl: "https://api.ensoul-ai.com/");
            Assert.That(config.ApiUrl, Is.EqualTo("https://api.ensoul-ai.com/v1"));
        }

        [Test]
        public void CustomHeaders_DefaultsToEmpty()
        {
            var config = new EnsoulConfig();
            Assert.That(config.CustomHeaders, Is.Not.Null);
            Assert.That(config.CustomHeaders, Is.Empty);
        }
    }
}
